### ISSUE TITLE
Can't set border-radius in custom themes if base theme is set to roundness 0

### ISSUE THREAD

**Issue by pmoleri** [IgniteUI/igniteui-angular#11603]
## Description  
Can't set border-radius in custom themes if base theme is set to roundness 0

 * igniteui-angular version: 13.1.15
 * browser: all

## Steps to reproduce  

1. Configure base theme with roundness 0:
```scss
@include core();
@include typography();
@include theme($palette: $palette, $schema: $schema, $roundness: 0);
```

2. Create custom theme:
```scss
$custom-avatar-theme: avatar-theme(
  $color: green,
  $border-radius: 5px,
);

:host {
  @include css-vars($custom-avatar-theme);
}
```

## Result  
Custom border-radius is ignored.

## Attachments  
![image](https://user-images.githubusercontent.com/12942794/169615652-1f0b0fbf-86d4-4e64-8eca-41e5780b1314.png)
https://stackblitz.com/edit/angular-muzqcx?file=src%2Fapp%2Fbuttons-style%2Fbuttons-style.component.scss

**Comment by pmoleri** [comment]
Hi @simeonoff @Lipata 
Any chance that we can get this in 13.1.x?
We worked around by setting roundness to 1 globally, and things look fine, but we risk missing some component affected by this global rounding.

**Comment by didimmova** [comment]
> Hi @simeonoff @Lipata 
> 
> Any chance that we can get this in 13.1.x?
> 
> We worked around by setting roundness to 1 globally, and things look fine, but we risk missing some component affected by this global rounding.

We already did https://github.com/IgniteUI/igniteui-angular/pull/11618

**Comment by pmoleri** [comment]
@didimmova @simeonoff 
I'm having another issue that looks related to this. When using custom theme for Avatar, roundShape stops working.
I updated the same stackblitz (using 13.1.17) and added some more cases for Avatar:
https://stackblitz.com/edit/angular-muzqcx?file=src/app/buttons-style/buttons-style.component.html
![image](https://user-images.githubusercontent.com/12942794/171046856-ccfeb950-1944-44cb-95fd-633138904a86.png)

**Comment by zdrawku** [comment]
@simeonoff could you please cherry-pick the fix to 13.1.x, we are still using it for AB and there would be a final release of 13.1.x next week.
